base/file.jl

```diff
@@ -420,13 +420,19 @@ const temp_prefix = "jl_"
 
 if Sys.iswindows()
 
+function _isroot(path::String)
+    m = match(r"^([^\\]+:|\\\\[^\\]+\\[^\\]+|\\\\\?\\UNC\\[^\\]+\\[^\\]+|\\\\\?\\[^\\]+:|)(.*)$", path)
+    m.captures[2] == "\\" ? true : false
+end
+
 function tempdir()
     temppath = Vector{UInt16}(undef, 32767)
     lentemppath = ccall(:GetTempPathW, stdcall, UInt32, (UInt32, Ptr{UInt16}), length(temppath), temppath)
     windowserror("GetTempPath", lentemppath >= length(temppath) || lentemppath == 0)
-    isroot = ccall((:PathIsRootW,:Shlwapi), stdcall, Int32, (Ptr{UInt16},), temppath)
-    isroot != 1 && temppath[lentemppath] == 0x005c ?  resize!(temppath,lentemppath-1) :  resize!(temppath,lentemppath)
-    return transcode(String, temppath)
+    resize!(temppath,lentemppath)
+    path = transcode(String, temppath)
+    # ignore trailing `\` except for drive-root
+    return _isroot(path) ?  path : path[end] == '\\' ? path[1:end-1] : path
```

Comment:
Isn't the common case for `temppath` to end with `\`, which means that this makes an extra copy of the string in the common case?

Reply:
Yea that is the common case. I'm not sure how'd go about avoiding the extra copy when using the julia  `_isroot` implementation.

Reply:
We could use a modified isroot regex that is missing the trailing slash, and test it on the string after we strip the slash—if it matches, add the slash back.